src/jvm/main/org/jetbrains/kotlinx/lincheck/TransformationClassLoader.java

```diff
@@ -120,7 +132,16 @@ public Class<?> loadClass(String name) throws ClassNotFoundException {
                 return result;
             }
             try {
-                byte[] bytes = instrument(originalName(name));
+                byte[] bytes = null;
+                if (bytecodeCache != null) {
+                    bytes = bytecodeCache.get(name);
```

Comment:
Please use `putIfAbscent(..) { .. }` instead

Reply:
Here the usage of `computeIfAbsent` in fact makes code more verbose, because:

1) `instrument` throws checked exception `IOException` which has to be handled inside lambda.
2) `bytecodeCache` might be null and it is Java file, so we cannot use Kotlin's `?` syntax sugar.

Reply:
Here is how the code looks like:

```java
                if (bytecodeCache != null) {
                    bytes = bytecodeCache.computeIfAbsent(name, className -> {
                            try {
                                return instrument(originalName(className));
                            } catch (IOException e) {
                                throw new RuntimeException(e);
                            }
                        }
                    );
                } else {
                    bytes = instrument(originalName(name));
                }
```

Reply:
We can remove checked exception from `instrument`, but still after this there would be two similar calls to `instrument` (in case when we have the cache, and when we don't)